opencast/opencast
# Refactoring workflows and encoding profiles

Based on and replaces #1504, as the original author cannot find the time to finish. For more details, see #1504.

To test this, Lars proposed this checklist:

<details>
  <summary>Plan for testing:</summary>
  
* Testing new workflows
 
   * [ ]  Upload presentation and publish directly
   * [ ]  Upload presenter and publish directly
   * [ ]  Upload two streams and publish directly
   * [ ]  Upload presentation, send to editor and publish from there
   * [ ]  Upload presenter, send to editor and publish from there
   * [ ]  Upload two streams, send to editor and publish from there
   * [ ]  Upload presentation, send to editor, cut out element and publish
   * [ ]  Upload presenter, send to editor, cut out element and publish
   * [ ]  Upload two streams, send to editor, cut out element and publish
   * [ ]  Retract event and publish it again
   * [ ]  Republish metadata
   * [ ]  Upload via studio (if the new workflow is used directly, needs more tests otherwise)

* Testing the upgrade path
 
   * [ ]  Upload presentation using old workflow, send to editor and publish from there using the new workflow
   * [ ]  Upload presenter using old workflow, send to editor and publish from there using the new workflow
   * [ ]  Upload two streams using old workflow, send to editor and publish from there using the new workflow
   * [ ]  Upload presentation using old workflow, send to editor, cut out element and publish using the new workflow
   * [ ]  Upload presenter using old workflow, send to editor, cut out element and publish using the new workflow
   * [ ]  Upload two streams using old workflow, send to editor, cut out element and publish using the new workflow
   * [ ]  Retract event published using old workflow and publish it again using the new workflow
   * [ ]  Republish metadata of event published using the old workflow

</details>

Regarding unanswered comments on the previous PR:
> I ran all the listed tests and only Retract event published using old workflow and publish it again using the new workflow does not work. Retracting from engage and publishing to engage again works. But the event is not retracted from the external api and oai-pmh.

The new workflows only publishes to or retracts from engage per default. Users are expected to add other operations to the workflows themselves, based on the publication channels they wish to support. So if you previously have published to the external api and oai-pmh, you should add the related retract operations to your retract workflow.

> Opencast has some encoding profiles hardcoded.

This PR adds any hardcoded profile that I found back to `opencast.properties`.
